### PR TITLE
feat (api): workflow build parameter and payload

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/worker/logger.go [hunk 1]
```diff
@@ -146,15 +151,28 @@ func (w *currentWorker) grpcLogger(inputChan chan sdk.Log) error {
...
-				log.Error("grpcLogger> Error sending message : %s", err)
-				//Close all
-				stream.CloseSend()
-				w.grpc.conn.Close()
-				w.grpc.conn = nil
-				//Reinject log
-				inputChan <- l
-				return nil
+			if w.currentJob.wJob == nil {
+				if err := stream.Send(&l); err != nil {
```
Comment (bnjjj) [inline engine/worker/logger.go]: Maybe make just `if` to choose the right struct is more readable and less duplications

File: engine/worker/logger.go [hunk 2]
```diff
@@ -146,15 +151,28 @@ func (w *currentWorker) grpcLogger(inputChan chan sdk.Log) error {
...
+					w.grpc.conn = nil
+					//Reinject log
+					inputChan <- l
+					return nil
+				}
+			} else {
+				if err := streamWorkflow.Send(&l); err != nil {
+					log.Error("grpcLogger> Error sending message : %s", err)
+					//Close all
+					stream.CloseSend()
```
Comment (bnjjj) [inline engine/worker/logger.go]: stream or streamWorkflow ? To be sure this isn't a copy/paste error

File: engine/api/workflow/execute_node_run.go [hunk 3]
```diff
@@ -311,5 +310,13 @@ func NodeBuildParameters(wf *sdk.Workflow, wr *sdk.WorkflowRun, id int64, u *sdk
 		}
 	}
 
+	if len(res) == 0 {
+		var err error
+		res, err = GetNodeBuildParameters(proj, wf, refNode, refNode.Context.DefaultPipelineParameters, refNode.Context.DefaultPayload)
+		if err != nil {
+			return nil, sdk.WrapError(sdk.ErrWorkflowNodeNotFound, "getWorkflowTriggerConditionHandler> Unable to get workflow node parameters")
```
Comment (yesnault) [inline engine/api/workflow/execute_node_run.go]: and add err in msg no ?

File: engine/api/workflow/process.go [hunk 4]
```diff
@@ -283,64 +268,57 @@ func processWorkflowNodeRun(db gorp.SqlExecutor, w *sdk.WorkflowRun, n *sdk.Work
...
-			m1 := sdk.ParametersToMap(r.Payload)
-			for k, v := range m1 {
-				m[k] = v
+			m1, errm1 := dump.ToMap(r.Payload, dump.WithDefaultLowerCaseFormatter())
+			if errm1 != nil {
+				AddWorkflowRunInfo(w, sdk.SpawnMsg{
+					ID:   sdk.MsgWorkflowError.ID,
+					Args: []interface{}{errm1},
+				})
+				log.Error("processWorkflowNodeRun> Unable to compute hook payload")
```
Comment (yesnault) [inline engine/api/workflow/process.go]: with errm1 in log, no ?

File: engine/api/workflow/process.go [hunk 5]
```diff
@@ -283,64 +268,57 @@ func processWorkflowNodeRun(db gorp.SqlExecutor, w *sdk.WorkflowRun, n *sdk.Work
...
+
+	//Process parameters for the jobs
+	//TODO inherit parameter from parent job
+	jobParams, errParam := getNodeRunBuildParameters(db, run)
+	if errParam != nil {
+		AddWorkflowRunInfo(w, sdk.SpawnMsg{
+			ID:   sdk.MsgWorkflowError.ID,
+			Args: []interface{}{errParam},
+		})
+		return errParam
```
Comment (yesnault) [inline engine/api/workflow/process.go]: no wrap ?

**Pull request merged**